visvis2/_wrappers.py

```diff
@@ -0,0 +1,204 @@
+import ctypes
+
+import wgpu
+import numpy as np
+
+
+def array_from_shader_type(spirv_type):
+    """ Get a numpy array object from a SpirV type from python-shader.
+    """
+    return np.asarray(spirv_type())
+
+
+# todo: can this be generic enough, keeping the GPU bits out / optional?
+
+# todo: Support for updating unmapped data. Use something like updateRange to do subBufferUpdate
+
+
+class BaseBufferWrapper:
+    """ A base buffer wrapper that can be implemented for numpy, ctypes arrays,
+    or any other kind of array.
+    """
+
+    def __init__(self, data=None, nbytes=None, usage=None, mapped=False):
```

Comment:
I see you've split this up into a base and a subclass for numpy based buffers. But aren't you missing ie. `itemsize` if you arrays are 1-dimensional?

Reply:
Yes, I think this makes a lot of sense.

Reply:
we can use numpy arrays and views to handle this. forcing shape convention is fine since there aren't many valid use cases (we can think of right now) for other conventions